src/include/enclave/enclave_oe.h

```diff
@@ -15,6 +18,26 @@ extern void* _dlstart_c(size_t base);
 
 extern int __libc_init_enclave(int argc, char** argv);
 
-sgxlkl_config_t* sgxlkl_enclave;
+sgxlkl_enclave_config_t* sgxlkl_enclave;
+
+typedef struct sgxlkl_enclave_disk_state
+{
+    size_t host_disk_index;
+    int fd;
+    char* mmap;
+    size_t capacity;
+    bool mounted;
+} sgxlkl_enclave_disk_state_t;
+
+typedef struct sgxlkl_enclave_state
+{
+    sgxlkl_enclave_config_t* enclave_config;
+    sgxlkl_app_config_t* app_config;
+
+    size_t num_disk_state;
+    sgxlkl_enclave_disk_state_t* disk_state;
+} sgxlkl_enclave_state_t;
```

Comment:
@prp I've started refactoring `sgxlkl_config_t sgxlkl_enclave` into this new config + state data structure which clearly separates the two. I had to touch a lot of lines of code related to the disks (and fix a couple bugs along the way) and I'm not sure whether this is what I should aim for. Do you think this type of refactoring is the right way to go? (Currently the host side is still unmodified and I build the new data structures up from the original ones, but I'm planning to refactor the host too once I'm done with the enclave. I can successfully run some EEID-enabled tests with this at this time.) BTW `sgxlkl_enclave_config_t` is what you call "the app config" and I'm planning to rename it once the existing `sgxlkl_app_config_t` isn't needed anymore.

Reply:
@wintersteiger this makes sense to me. Currently the separation between `sgxlkl_enclave_config_t` and `sgxlkl_app_config_t` is not clear, and there is overlap between the two.

By the way, I keep using the term `app_config` for legacy reasons, but it doesn't describe well that this includes the full configuration of the enclave environment that the application will find. Perhaps now the time is ripe to use a more sensible name in the code?

Reply:
Yeah, that's why I started typing `struct name_t` when I need to be clear about which one I mean. The new, combined, clean config is called `sgxlkl_enclave_config_t` in this branch and `sgxlkl_enclave` is of this new type (instead of `sgxlkl_config_t`). It currently contains variables with the same names as the `sgxlkl_config_t`, so that the code can refer to the same names, so that I can keep compiling/testing while refactoring. It doesn't contain anything named app_config, but we can rename it once the current `sgxlkl_app_config_t` isn't needed anymore (currently it's still used on the host).
This code may be really hard to review if you don't have perfect knowledge of the name-mapping in my brain. I hope it will make sense when I'm done :-)